src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/PairHMMLikelihoodCalculationEngine.java

```diff
@@ -381,12 +385,12 @@ void applyPCRErrorModel( final byte[] readBases, final byte[] readInsQuals, fina
                 bestRepeatUnit = bestFWRepeatUnit; // arbitrary
             } else {
                 // tandem repeat starting forward from current offset.
-                // It could be the case that best BW unit was differnet from FW unit, but that BW still contains FW unit.
+                // It could be the case that best BW unit was different from FW unit, but that BW still contains FW unit.
                 // For example, TTCTT(C) CCC - at (C) place, best BW unit is (TTC)2, best FW unit is (C)3.
                 // but correct representation at that place might be (C)4.
                 // Hence, if the FW and BW units don't match, check if BW unit can still be a part of FW unit and add
                 // representations to total
-                maxBW = GATKVariantContextUtils.findNumberOfRepetitions(bestFWRepeatUnit, Arrays.copyOfRange(readBases, 0, offset + 1), false);
+                maxBW = GATKVariantContextUtils.findNumberOfRepetitions(bestFWRepeatUnit, testString, false);
```

Comment:
It seems like you're potentially doing an extra copy by pulling `testString` out of this block.  Why not initialize it here in the `else` block?